sklearn/semi_supervised/_label_propagation.py

```diff
@@ -190,7 +190,7 @@ class labels.
         """
         check_is_fitted(self)
 
-        X_2d = check_array(X, accept_sparse=['csc', 'csr', 'coo', 'dok',
+        X_2d = check_array(X, accept_sparse=['csc', 'csr', 'coo',
```

Comment:
should this use `_validate_data`? ping @NicolasHug

Reply:
I'm also really not sure why this fixes the issue

Reply:
> should this use _validate_data

We only use it in `fit` for now, so `check_array` is fine here

Reply:
Thanks for the feedback @adrinjalali: PR #17085 already contains tests extended to the sparse case. Could you please advise a newbie regarding regression tests?